src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncIteratorTests.cs

```diff
@@ -4140,6 +4144,105 @@ async Unknown local()
                 );
         }
 
+        [ConditionalFact(typeof(WindowsDesktopOnly))]
+        public void DisposeAsyncInBadState()
+        {
+            string source = @"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    static async System.Collections.Generic.IAsyncEnumerable<int> M(CancellationToken token)
+    {
+        yield return 1;
+        while (true)
+        {
+            await Task.Delay(10);
```

Comment:
Why not `Task.Yield`?

Reply:
Habit and didn't know about `Yield()`. The main goal is to avoid the optimization for already completed tasks.

---
In reply to: [244785798](https://github.com/dotnet/roslyn/pull/31764#discussion_r244785798) [](ancestors = 244785798)

Reply:
Understood. The problem with `Delay` is that over the totality of our unit tests it ends up causing a noticable performance issue. At least historically it's done that.